src/Wallabag/CoreBundle/Controller/ExportController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Wallabag\CoreBundle\Controller;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Wallabag\CoreBundle\Entity\Entry;
+use Wallabag\CoreBundle\Helper\EntriesExport;
+
+class ExportController extends Controller
+{
+    /**
+     * Gets all entries for current user.
+     *
+     * @Route("/export/{category}.{format}", name="ebook", requirements={
+     *     "_format": "epub|mobi|pdf|json|xml|txt|csv"
+     * })
+     */
+    public function getEntriesAction($format, $category)
+    {
+        $repository = $this->getDoctrine()->getRepository('WallabagCoreBundle:Entry');
+        switch ($category) {
```

Comment:
Instead of this big switch / case, you can define what `category` can be in the url (the same way you did for `_format` and then you just need to apply a `ucfirst()` to have the `$method`.